app/javascript/contentDisplayPolicy/initHiddenComments.js

```diff
@@ -42,20 +16,65 @@ For further actions, you may consider blocking this person and/or reporting abus
       });
   }
 
+  function showHideCommentsModal(commentId) {
+    const form = document.getElementById('hide-comments-modal__form');
+    form.action = `/comments/${commentId}/hide`;
+
+    window.Forem.showModal({
+      title: 'Confirm hiding the comment',
+      contentSelector: '#hide-comments-modal',
+      overlay: true,
+    }).then(() => {
+      const hideCommentForms = document.querySelectorAll(
+        '.hide-comments-modal__form',
+      );
+
+      hideCommentForms.forEach((form) => {
+        form.addEventListener('submit', handleHideCommentsFormSubmit);
+      });
+    });
+  }
+
   const hideLinks = Array.from(document.getElementsByClassName('hide-comment'));
 
   hideLinks.forEach((link) => {
     const { hideType, commentId } = link.dataset;
-
     if (hideType === 'hide') {
-      link.addEventListener('click', () => {
-        hide(commentId);
+      link.addEventListener('click', (e) => {
+        e.preventDefault();
+        showHideCommentsModal(commentId);
       });
     } else if (hideType === 'unhide') {
       link.addEventListener('click', () => {
         unhide(commentId);
       });
     }
   });
+
+  const handleHideCommentsFormSubmit = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    const { target: form } = e;
+    const hide_children_check = form.getElementsByClassName('hide_children')[0];
+    let url = form.action;
+
+    if (hide_children_check.checked) {
+      url = `${url}?hide_children=1`;
+    }
```

Comment:
(just a suggestion / not blocking) - I think this would do the same (it's down to personal preference though so feel free to ignore)

```suggestion
const url = `${form.action}${hide_children_check.checked ? '?hide_children=1' : ''}`
```